### PR TITLE
virt-launcher, sriov: Update the domain config state on detach

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

When detaching a SR-IOV host-device, the existing `DetachDevice`
function only updates the domain live state.

In order to update the domain config, the `DetachDeviceFlags` needs to
be used and instructed to update both domain states.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This should be a per-requirement for #5037 .

The issue has pop up after #5010 got in, fixing the domain mode on the target (from transient to persisted).

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
@jean-edouard , @awels , you may find this relevant to the current usages of `DetachDevice` in the code (detaching disks).
I think any usage of `DetachDevice` needs to be replaced with `DetachDeviceFlags`.

**Comment by awels** [comment]
What about AttachDevice? I am not sure I understand what is happening here, so some context might be helpful. I am definitely calling Attach/DetachDevice for hotplugging disks, so this is likely relevant, I just don't know enough about what changes here.

**Comment by EdDev** [comment]
> What about AttachDevice? I am not sure I understand what is happening here, so some context might be helpful. I am definitely calling Attach/DetachDevice for hotplugging disks, so this is likely relevant, I just don't know enough about what changes here.

Yes, both have the same problem, the later is handled by #5037 .
They update the domain *live* state and not the *config* one. That is problematic, because kubevirt uses persisted domains, which means we must update the config as well.

**Comment by awels** [comment]
Okay, so hotplug attach and detach device should look something like:
```go
DetachDeviceFlags(string(devXML), libvirt.DOMAIN_DEVICE_MODIFY_LIVE|libvirt.DOMAIN_DEVICE_MODIFY_CONFIG)
AttachDeviceFlags(string(devXML), libvirt.DOMAIN_DEVICE_MODIFY_LIVE|libvirt.DOMAIN_DEVICE_MODIFY_CONFIG)
```
Which will update both live and config. Once #5152 and #5307 have merged I can make a PR to fix the hotplug calls.

**Comment by ormergi** [comment]
Verified on local setup also with #5037 sriov tests are passing

**Review by ormergi — APPROVED** [review]
Thanks!

**Comment by qinqon** [comment]
"""
/hold
until emergency CI fixes of #5207 are merged
"""

**Comment by EdDev** [comment]
> """
> /hold
> until emergency CI fixes of #5207 are merged
> """

5207 got merged.
/hold cancel

**Pull request merged**
